kong2kic/route.go

```diff
@@ -256,3 +259,288 @@ func addPluginsToRoute(
 	}
 	return nil
 }
+
+// HeadersByNameTypeAndValue is a type to sort headers by name, type and value.
+// This is needed to ensure that the order of headers is consistent across runs.
+type HeadersByNameTypeAndValue []k8sgwapiv1.HTTPHeaderMatch
+
+func (a HeadersByNameTypeAndValue) Len() int      { return len(a) }
+func (a HeadersByNameTypeAndValue) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+func (a HeadersByNameTypeAndValue) Less(i, j int) bool {
+	if a[i].Name < a[j].Name {
+		return true
+	}
+	if a[i].Name > a[j].Name {
+		return false
+	}
+
+	if a[i].Type != nil && a[j].Type == nil {
+		return true
+	}
+	if a[i].Type == nil && a[j].Type != nil {
+		return false
+	}
+
+	if *a[i].Type < *a[j].Type {
+		return true
+	}
+	if *a[i].Type > *a[j].Type {
+		return false
+	}
+	return a[i].Value < a[j].Value
+}
+
+// Convert route to HTTPRoute (Gateway API)
+func populateKICIngressesWithGatewayAPI(content *file.Content, kicContent *KICContent) error {
+	for _, service := range content.Services {
+		for _, route := range service.Routes {
+			httpRoute, err := createHTTPRoute(service, route)
+			if err != nil {
```

Comment:
Just curious: why are we continuing here even with the err?

Reply:
The input Kong configuration may have many routes that must be converted to HTTPRoute objects. There may be an error when converting one of the routes. If that is the case, we inform of the error and continue processing the next route object.